strapi/strapi
# Solve foreign key error on DEITS

### What does it do?

It handles the foreign key constraint error showing a warning in the server logs instead of throwing an error and stopping the transfer process.

### Why is it needed?

Users complain that their processes stop because of foreign key constraint errors and we've detected that even though it's potentially due to inconsistencies in their DBs, it's not worth throwing and error and it will be enough to handle these situations with a warning.

### How to test it?

Do a transfer with orphans in your database (create an entity with a relation to another content type and in the database delete one of the ends of the relation, leaving the row linking both content types in the DB). Then execute a transfer to another Strapi project and observe that there is a warning in the server about the foreign key constraint, but the transfer continues as expected. 

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/18350
